src/core/node.jl

```diff
@@ -227,13 +224,29 @@ function _after_construct_variables!(node::Node)
     # We can now properly finalize the `state_lb`, and `state_ub`.
     _finalize(node.state_lb)
     _finalize(node.state_ub)
+    _finalize(node.state_initial)
+    _finalize(node.state_final)
+    # Check expressions
+    if node.state_initial.value isa Vector
+        @critical "[build] The initial value of a Node must be scalar." component = node.name
+    end
+    if node.state_final isa Vector
+        @critical "[build] The final value of a Node must be scalar." component = node.name
+    end
+    if node.state_cyclic !== :disabled && !isnothing(node.state_initial.value) && !isnothing(node.state_initial.value)
```

Comment:
```suggestion
    if node.state_cyclic !== :disabled && !_isempty(node.state_initial) && !_isempty(node.state_initial)
```

I know that's probably not really solid, but the idea behind that was that there could be different reasons for an expression being "empty", not limited to the value being `nothing` (and maybe `nothing` could even be "non-empty" for specific cases).

See [_isempty](https://github.com/ait-energy/IESopt.jl/blob/463763b502a4dc09cd32c8a46b772031ba85ae7f/src/core/expression.jl#L265), which I hope to be equivalent in this case (?) - due to this [`nothing`-to-expression conversion](https://github.com/ait-energy/IESopt.jl/blob/463763b502a4dc09cd32c8a46b772031ba85ae7f/src/core/expression.jl#L286).